File: 1.5-ArrayBasics/ArrayBasics_in_CPP/3-SwapAlternate.cpp
#include<bits/stdc++.h>
using namespace std;

void printArray(int arr[], int n){
    for(int i=0; i<n; i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}

void swap(int& a, int& b){
    int temp = a;
    a = b;
    b = temp;
}

void alternate(int arr[], int n){
    int i = 0;

    while(i+1 < n){
        swap(arr[i],arr[i+1]);
        i += 2;
    }
}

int main() {

    int arr[5] = {1,2,3,4,5};
    int brr[6] = {1,2,3,4,5,6};

    alternate(arr,5);
    alternate(brr,6);

    printArray(arr,5);
    printArray(brr,6);

    return 0;
}

File: 1.5-ArrayBasics/ArrayBasics_in_Python/3-SwapAlternate.py
def printArray(arr):
    for i in range(len(arr)):
        print(arr[i], end=" ")
    print('\n')

def swapAlternate(arr):
    i = 0

    while(i+1<len(arr)):
        arr[i], arr[i+1] = arr[i+1], arr[i]
        i += 2


arr = [1,2,3,4,5,6]
brr = [1,2,3,4,5]

swapAlternate(arr)
swapAlternate(brr)

printArray(arr)
printArray(brr)